SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -24,47 +30,106 @@
 ASpatialTestRepNotify::ASpatialTestRepNotify()
 	: Super()
 {
-	Author = "Miron + Andrei";
+	Author = "Miron + Andrei + Arthur";
 	Description = TEXT("Test RepNotify replication and shadow data");
 }
 
 void ASpatialTestRepNotify::PrepareTest()
 {
 	Super::PrepareTest();
 
-	// The Server sets some initial values to the replicated variables.
-	AddStep(TEXT("SpatialTestRepNotifyServerSetReplicatedVariables"), FWorkerDefinition::Server(1), nullptr, [this]() {
-		OnChangedRepNotifyInt1 = 1;
-		AlwaysRepNotifyInt1 = 2;
-		OnChangedRepNotifyInt2 = 3;
-		AlwaysRepNotifyInt2 = 4;
-		TestArray.Empty();
-		TestArray.Add(1);
-		TestArray.Add(2);
+	/**
+	 * Here we check core rep notify functionality and the order rep notifies are called in.
+	 * In native, data will be applied for an actor, its subobjects, and only THEN will repnotifies be called for the updated data.
+	 *
+	 * To test this, we set the update a replicated property on an actor and a subobject of its.
+	 * Then in repnotifies on both the actor and the subobject, we check that the property on the other object has already been updated.
+	 * */
+	AddStep(TEXT("SpatialTestRepNotifyServerSetReplicatedVariables1"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		TestActor = GetWorld()->SpawnActor<ASpatialTestRepNotifyActor>(FVector(0, 0, 0), FRotator::ZeroRotator, FActorSpawnParameters());
+		RegisterAutoDestroyActor(TestActor);
+
+		TestActor->TestSubobject->OnChangedRepNotifyInt = 400;
+		TestActor->OnChangedRepNotifyInt1 = 350;
+
```

Comment:
@samiwh This is the relevant bit. Would these all go in the same bunch (or whatever we call it) to runtime?

Reply:
this should all go into one CreateEntityRequest and therefore we should be safe

Reply:
kool